websocket.go

```diff
@@ -207,7 +229,11 @@ type Client struct {
 	service     string
 	connections map[destination]*websocket.Conn
 	suite       network.Suite
-
+	// whether we should use TLS
+	tls bool
+	// Pool of certificates we trust, can be used for self-signed certificates;
+	// only useful if tls is set to true.
+	trustedCertificates *x509.CertPool
```

Comment:
Same as the server side:

This makes it impossible for callers to change the tls.Config if they want to. Instead, give them a TLSClientConfig field they can set, and then in Send(), do `d := &websocket.Dialer{ TLSClientConfig: c.TLSClientConfig }`